helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -457,13 +819,23 @@ public void onLiveInstanceChange(List<LiveInstance> liveInstances,
   @PreFetch(enabled = false)
   public void onStateChange(String instanceName, List<CurrentState> statesInfo,
       NotificationContext changeContext) {
-    if (_sourceDataType.equals(PropertyType.CURRENTSTATES)) {
+    if (_sourceDataTypeMap.containsKey(PropertyType.CURRENTSTATES)) {
       _routerUpdater.queueEvent(changeContext, ClusterEventType.CurrentStateChange,
           HelixConstants.ChangeType.CURRENT_STATE);
     } else {
       logger.warn(
           "RoutingTableProvider does not use CurrentStates as source, ignore CurrentState changes!");
     }
+    _routerUpdater.queueEvent(changeContext, ClusterEventType.CurrentStateChange,
+        HelixConstants.ChangeType.CURRENT_STATE);
+  }
+
+  @Override
+  @PreFetch(enabled = false)
+  public void onCustomizedViewChange(List<CustomizedView> customizedViewList,
+      NotificationContext changeContext) {
+    _routerUpdater.queueEvent(changeContext, ClusterEventType.CustomizedViewChange,
```

Comment:
Safer to add the same source type check we did for the other listeners.

Reply:
Done.

Reply:
Done.